### ISSUE TITLE
AudioDevice recording flattens stereo audio into mono

### ISSUE THREAD

**Issue by visualmusicdesign** [videokit-ai/videokit#159]
I recently got more emails from users complaining that audio recording on MacOS (which on my app is through AudioDevice in VideoKit) is flattening the stereo channels into a single mono channel, copied to both L + R.

Here is the code used to start recording:

`audioRecordingDevice = devices[selectedAudioInput];`

`audioRecordingDevice.channelCount = 2;`

`clipOutput = new AudioClipOutput();`

`audioRecordingDevice.StartRunning(clipOutput);`

I had corresponded with you about this issue way back in 2020 in NatDevice, at which time you said you would need to table this issue as you were unable to find a solution.

It remains a serious issue - I’m attaching the files demonstrating the issue my user sent me.  One is recorded through Quicktime, the other is recorded through AudioDevice in VideoKit.  
[https://www.dropbox.com/scl/fo/v377hzcmpig4ur3pizbs5/AEOXj54Gp1gl7SMXjZMScww?rlkey=by5n2sbedy3kex5y3ms9acnm7&dl=0](url)

You can hear that the AudioDevice file is essentially recording the stereo audio as mono audio, copied to both channels - which sounds lower quality and lower depth.

Is there any chance of this getting fixed in a future VideoKit update?  

Happy to test anytime.

Thank you

**Comment by olokobayusuf** [comment]
Hey @visualmusicdesign seems like we could reimplement how we set the audio format on macOS to improve on this issue (it's technically not a bug, as the OS decides what to do here). I'll have updates next week.

**Comment by olokobayusuf** [comment]
@visualmusicdesign I've spent quite a bit of time on this issue, and it seems that on macOS specifically `AVFoundation` is limited for hi-fi audio. Audio devices correctly report their supported formats (`AVCaptureDeviceFormat`), but we can't configure the streaming `AVCaptureSession` to keep the correct format we've selected.

It looks like the [relevant configuration constant](https://developer.apple.com/documentation/avfoundation/avcapturesession/preset/inputpriority?language=objc) is available on iOS, but not macOS.
